cpp/src/parquet/column_reader.cc

```diff
@@ -1328,6 +1329,156 @@ class TypedRecordReader : public ColumnReaderImplBase<DType>,
 
     return records_read;
   }
+  
+  // Skip records that we have in our buffer. This function is only for
+  // non-repeated fields.
+  int64_t SkipRecordsInBufferNonRepeated(int64_t num_records) {
+    ARROW_DCHECK(this->max_rep_level_ == 0);
+    ARROW_DCHECK(this->has_values_to_process());
```

Comment:
The first DCHECK seems self explanitory, I'm not sure I understand the second one though.  Just to validate, do you think DCHECK or a throwing here is more appropriate?

Reply:
I removed the check and added a return statement.